schema-registry/rest-service/src/test/java/com/hortonworks/registries/schemaregistry/avro/SchemaBranchLifeCycleTest.java

```diff
@@ -89,293 +86,245 @@ public SchemaBranchLifeCycleTest(SchemaRegistryTestProfileType schemaRegistryTes
+        SchemaMetadata schemaMetadata = addSchemaMetadata(testNameRule.getMethodName(), SchemaCompatibility.NONE);
+
+        SchemaIdVersion masterSchemaIdVersion1 = addSchemaVersion(SchemaBranch.MASTER_BRANCH, schemaMetadata, "/device.avsc");
+        SchemaBranch schemaBranch1 = addSchemaBranch("BRANCH1", schemaMetadata, masterSchemaIdVersion1.getSchemaVersionId());
+        SchemaIdVersion schemaBranch1Version1 = addSchemaVersion(schemaBranch1.getName(), schemaMetadata, "/device-incompat.avsc");
 
-        SchemaBranch persistedSchemaBranch = schemaRegistryClient.createSchemaBranch(schemaIdVersion1.getSchemaVersionId(), branch1);
+        schemaRegistryClient.enableSchemaVersion(schemaBranch1Version1.getSchemaVersionId());
+        schemaRegistryClient.archiveSchemaVersion(schemaBranch1Version1.getSchemaVersionId());
 
-        String schema2 = AvroSchemaRegistryClientUtil.getSchema("/device-incompat.avsc");
-        SchemaIdVersion schemaIdVersion = schemaRegistryClient.addSchemaVersion(branch1.getName(), schemaMetadata.getName(), new SchemaVersion(schema2, "second version"));
+        schemaRegistryClient.deleteSchemaBranch(schemaBranch1.getId());
+    }
+
+
+    @Test (expected = InvalidSchemaBranchDeletionException.class)
+    public void deleteBranchWithRootVersionOfAnotherBranch() throws InvalidSchemaException, SchemaNotFoundException, IncompatibleSchemaException, IOException, SchemaBranchAlreadyExistsException, SchemaLifecycleException, InvalidSchemaBranchDeletionException {
+        SchemaMetadata schemaMetadata = addSchemaMetadata(testNameRule.getMethodName(), SchemaCompatibility.NONE);
+
+        SchemaIdVersion masterSchemaIdVersion1 = addSchemaVersion(SchemaBranch.MASTER_BRANCH, schemaMetadata, "/device.avsc");
+        SchemaBranch schemaBranch1 = addSchemaBranch("BRANCH1", schemaMetadata, masterSchemaIdVersion1.getSchemaVersionId());
+        SchemaIdVersion schemaBranch1Version1 = addSchemaVersion(schemaBranch1.getName(), schemaMetadata, "/device-incompat.avsc");
+        addSchemaBranch("CHILD-BRANCH", schemaMetadata,schemaBranch1Version1.getSchemaVersionId());
 
-        String schema3 = AvroSchemaRegistryClientUtil.getSchema("/device-compat.avsc");
-        schemaRegistryClient.addSchemaVersion(branch1.getName(), schemaMetadata.getName(),
-                new SchemaVersion(schema3, "second version"));
+        schemaRegistryClient.enableSchemaVersion(schemaBranch1Version1.getSchemaVersionId());
+        schemaRegistryClient.archiveSchemaVersion(schemaBranch1Version1.getSchemaVersionId());
 
-        schemaRegistryClient.enableSchemaVersion(schemaIdVersion.getSchemaVersionId());
+        schemaRegistryClient.deleteSchemaBranch(schemaBranch1.getId());
+    }
 
-        schemaRegistryClient.deleteSchemaBranch(persistedSchemaBranch.getId());
+    @Test (expected = RuntimeException.class)
```

Comment:
Good to have  github issue filed for returning `SchemaLifecycleException` instead of `RuntimeException` while deleting a root schema version of a branch.

Reply:
Filed #386